src/app/clusters/service-area-server/service-area-array-attributes-delegate.h

```diff
@@ -0,0 +1,88 @@
+/*
+ *
+ *    Copyright (c) 2024 Project CHIP Authors
+ *    All rights reserved.
+ *
+ *    Licensed under the Apache License, Version 2.0 (the "License");
+ *    you may not use this file except in compliance with the License.
+ *    You may obtain a copy of the License at
+ *
+ *      
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *    Unless required by applicable law or agreed to in writing, software
+ *    distributed under the License is distributed on an "AS IS" BASIS,
+ *    WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *    See the License for the specific language governing permissions and
+ *    limitations under the License.
+ */
+
+#pragma once
+
+
+#include "service-area-cluster-objects.h"
+
+
+namespace chip {
+namespace app {
+namespace Clusters {
+namespace ServiceArea {
+
+
+
+/**
+ * A class which wraps a Service Area Cluster LocStruct instance, adding string storage and string utility functions.
+ */
+class ServiceAreaArrayAttributeDelegate
+{
+private:  // temporary
+
+    LocationStructureWrapper  mSupportedLocation {  777, 
+                                                    DataModel::Nullable<uint_fast8_t>(),
+                                                    CharSpan::fromCharString("My Location D"),
+                                                    DataModel::Nullable<int16_t>( 1),
+                                                    DataModel::Nullable<ServiceArea::AreaTypeTag>(),
+                                                    DataModel::Nullable<ServiceArea::LandmarkTag>(ServiceArea::LandmarkTag::kCouch),
+                                                    DataModel::Nullable<ServiceArea::PositionTag>(ServiceArea::PositionTag::kNextTo),
+                                                    DataModel::Nullable<ServiceArea::FloorSurfaceTag>(ServiceArea::FloorSurfaceTag::kHardwood) };
+protected:
+    ServiceAreaArrayAttributeDelegate() {}
+    ~ServiceAreaArrayAttributeDelegate() = default;
+
+private:
+    ServiceAreaArrayAttributeDelegate(const ServiceAreaArrayAttributeDelegate &)             = delete;
+    ServiceAreaArrayAttributeDelegate & operator=(const ServiceAreaArrayAttributeDelegate &) = delete;
+
+
+public:
+
+//*************************************************************************
+// Supported Locations accessors
+
+virtual size_t GetNumberOfSupportedLocations() = 0;
+
+virtual bool GetSupportedLocationByIndex(uint32_t listIndex, const LocationStructureWrapper *& supportedLocation) = 0;
```

Comment:
Why is `cost *&` used here, and everywhere else in general? Shouldn't this be `&`? It seems that you are intending for the delegate to hold a permanent and unchanging list of locations. Consider the scenario where this function returns a pointer to a valid location but that the list of supported locations changes is some way.

Reply:
I'm using const as the server, the delegate, and the application should have very limited write access to the location contents.  Modifications which are allowed are intended to be handled through modify API's.

Will be changing these to objects that get copied from the attribute (change *& to &), in which case they will necessarily be non-const, and modifying the object won't change the attribute list (unless the object end up getting passed back somehow.)